application/frontend/controllers/MfaController.php

```diff
@@ -0,0 +1,138 @@
+<?php
+namespace frontend\controllers;
+
+use common\components\MfaBackendInterface;
+use common\models\Mfa;
+use common\models\User;
+use frontend\components\BaseRestController;
+use yii\web\BadRequestHttpException;
+use yii\web\NotFoundHttpException;
+use yii\web\ServerErrorHttpException;
+
+class MfaController extends BaseRestController
+{
+
+    /**
+     * Create new MFA record
+     * @return array
+     * @throws BadRequestHttpException
+     */
+    public function actionCreate()
+    {
+        $req = \Yii::$app->request;
+        $type = $req->getBodyParam('type','invalid');
+        if ( ! Mfa::isValidType($type)) {
+            throw new BadRequestHttpException('The provided type is not a supported MFA type', 1506695647);
+        }
+
+        $employeeId = $req->getBodyParams('employee_id');
```

Comment:
Looks like this `getBodyParams(...)` was meant to be singular ([`getBodyParam(...)`](http://www.yiiframework.com/doc-2.0/yii-web-request.html#getBodyParam()-detail)).

Reply:
yep, interesting it worked